editors/code/src/binary.ts

```diff
@@ -15,22 +17,38 @@ export async function resolveAirBinaryPath(
 		return BUNDLED_AIR_EXECUTABLE;
 	}
 
+	// User requested the `"bundled"` air binary
 	if (executableLocation === "bundled") {
-		// User requested the `"bundled"` air binary
-		output.log(
-			`Using bundled executable as requested by \`air.executableLocation\`: ${BUNDLED_AIR_EXECUTABLE}`,
-		);
-		return BUNDLED_AIR_EXECUTABLE;
-	} else {
-		// User requested `"environment"`, so check the `PATH` first
-		const environmentPath = await which(AIR_BINARY_NAME, { nothrow: true });
-
-		if (environmentPath) {
-			output.log(`Using environment executable: ${environmentPath}`);
-			return environmentPath;
-		} else {
-			output.log(`Using bundled executable: ${BUNDLED_AIR_EXECUTABLE}`);
+		if (fs.existsSync(BUNDLED_AIR_EXECUTABLE)) {
+			output.log(
+				`Using bundled executable as requested by \`air.executableLocation\`: ${BUNDLED_AIR_EXECUTABLE}`,
+			);
 			return BUNDLED_AIR_EXECUTABLE;
 		}
+
+		// Fallthrough
+		output.log(`Bundled executable not found: ${BUNDLED_AIR_EXECUTABLE}`);
```

Comment:
```suggestion
		output.log(`Bundled executable not found, falling back to environment: ${BUNDLED_AIR_EXECUTABLE}`);
```

Makes the flow a little more self explanatory I think